### DISCUSSION TITLE
Indenting word wrapped lines while editing?

### DISCUSSION THREAD

**Discussion by virtuallyvlad** [standardnotes/forum#2594] [Suggestions]
Can any of the editors indent word wrapped lines while in edit mode?

Meaning, if you start a line indented and it ends up wrapping to the next line, the next line is indented the same amount (or, optionally, one level more).

I only saw this feature in the preview/view modes of a couple, whereas most desktop notepads allow this while editing.  E.g., Notepad++ calls it "Aligned Line Wrap" or "Indent Line Wrap", and Notepad2 allows you to "Wrap indent as first subline" or "Wrap indent 1 level more than first subline".

It's a feature that really improves the legibility and organization of a note when applying hierarchical or nested indentation.  I think it's also relatively common for code in order to see logical structures more easily.

**Comment by atmoio** [comment]
You might check the Code Editor, Vim Editor, or Minimal Markdown Editors for this. If they aren't supported in any of those, it's unlikely we'll add them, unless you can find some documentation for CodeMirror that states this would be possible.

**Comment by virtuallyvlad** [comment]
Thanks for the reply.  I do not see any of those three indenting wrapped text.

I checked the others again and these do it but in preview mode only:
- Simple Markdown Editor
- Fancy Markdown Editor
- Advanced Markdown Editor

There is this CodeMirror demo which does it by changing the renderLine event:
https://codemirror.net/demo/indentwrap.html

And another CodeMirror demo which uses the above and also detects "-", ">", "1." as part of identation:
https://codepen.io/cben/pen/JdKyRE

Just for reference, there are also other embedded javascript editors with this: Ace, Atom, and Brackets (based on CodeMirror).  Understood that another Extension is probably not the highest priority :)

**Comment by atmoio** [comment]
Thanks, I'll check this out when I get the chance. If you happen to find a working solution, feel free to submit a PR to any of the editors, located [here](https://github.com/sn-extensions).

**Comment by virtuallyvlad** [comment]
Just an update for anyone looking to have nicely-wrapped lines _while editing_: the following three editors do it, and come with some other useful features:
- [Indent Editor](https://github.com/MaxLap/standard-notes-indent-editor)
- [SN Nimble Editor](https://hub.darcs.net/jandrew/sn-nimble-editor) - based off Indent Editor
- [Fast Editor](https://github.com/dec0dOS/standard-notes-fast-editor) - based off Indent Editor

And a related request:
https://github.com/standardnotes/forum/issues/182
